sql/sql_analyze_stmt.h

```diff
@@ -51,10 +84,20 @@ class Exec_time_tracker
   ulonglong count;
   ulonglong cycles;
   ulonglong last_start;
+  /*
+    The time spent between stop_tracking() call on this object and any
+    other time measurement will be billed to this tracker.
+  */
+  Gap_time_tracker *my_gap_tracker;
 
+  ulonglong measure() const
+  {
+    return MY_TIMER_ROUTINE_CYCLES ? my_timer_cycles()
```

Comment:
I could use a `#if MY_TIMER_ROUTINE !=0` condition since `MY_TIMER_ROUTINE` is a define if you think its more readable.